src/Unidevel.OpenWeather/OpenWeatherClient.cs

```diff
@@ -9,34 +9,46 @@ public class OpenWeatherClient : IDisposable, IOpenWeatherClient
     {
         private readonly IConfiguration _configuration;
         private readonly string _apiKey;
+        private bool _disposed = false;
+
+        private readonly HttpClient _httpClient = new HttpClient();
 
         public OpenWeatherClient(IConfiguration configuration = null, string apiKey = null)
         {
             _configuration = configuration;
-            _apiKey = apiKey ?? configuration?.GetSection("OpenWeather")?["ApiKey"];            
+            _apiKey = apiKey ?? configuration?.GetSection("OpenWeather")?["ApiKey"];
         }
 
         public async Task<CurrentWeather> GetCurrentWeatherAsync(float longitude = float.NaN, float latitude = float.NaN, string cityNameCountryCode = null, int cityId = Int32.MinValue, string apiKey = null)
         {
             var url = buildCurrentWeatherUrl(apiKey ?? _apiKey, longitude, latitude, cityNameCountryCode, cityId);
 
-            string currentWeatherJson;
-            using (var webClient = new WebClient()) currentWeatherJson = await webClient.DownloadStringTaskAsync(url);
-
-            var currentWeather = Newtonsoft.Json.JsonConvert.DeserializeObject<CurrentWeather>(currentWeatherJson);
-
-            return currentWeather;
+            try
+            {
+                var response = await _httpClient.GetStringAsync(url);
+                var currentWeather = Newtonsoft.Json.JsonConvert.DeserializeObject<CurrentWeather>(response);
+                return currentWeather;
+            }
+            catch (HttpRequestException)
+            {
+                throw;
+            }
         }
 
         public async Task<WeatherForecast> GetWeatherForecast5d3hAsync(float longitude = float.NaN, float latitude = float.NaN, string cityNameCountryCode = null, int cityId = Int32.MinValue, string apiKey = null)
         {
             var url = buildWeatherForecast5d3hUrl(apiKey ?? _apiKey, longitude, latitude, cityNameCountryCode, cityId);
 
-            string weatherForecastJson;
-            using (var webClient = new WebClient()) weatherForecastJson = await webClient.DownloadStringTaskAsync(url);
-            var weatherForecast = Newtonsoft.Json.JsonConvert.DeserializeObject<WeatherForecast>(weatherForecastJson);
-
-            return weatherForecast;
+            try
+            {
+                var response = await _httpClient.GetStringAsync(url);
+                var weatherForecast = Newtonsoft.Json.JsonConvert.DeserializeObject<WeatherForecast>(response);
+                return weatherForecast;
+            }
+            catch (HttpRequestException)
+            {
+                throw;
+            }
         }
 
         private static readonly IFormatProvider americanNumberFormatProvider = System.Globalization.CultureInfo.GetCultureInfo("en-US").NumberFormat;
```

Comment:
Probably different naming scheme should be used. PascalCase instead of camelCase. Verify it. camelCase was fine few years ago.